src/sonic-daemon-common/sonic_daemon_common/daemon_common.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python2
+
+try:
+    import imp
+    import signal
+    import subprocess
+    import sys
+    import syslog
+except ImportError, e:
+    raise ImportError (str(e) + " - required module not found")
+
+#============================= Constants =============================
+
+# Platform root directory inside docker
+PLATFORM_ROOT_DOCKER = "/usr/share/sonic/platform"
+SONIC_CFGGEN_PATH = '/usr/local/bin/sonic-cfggen'
+HWSKU_KEY = 'DEVICE_METADATA.localhost.hwsku'
+PLATFORM_KEY = 'DEVICE_METADATA.localhost.platform'
+
+# Redis DB information
+REDIS_HOSTNAME = "localhost"
+REDIS_PORT = 6379
+REDIS_TIMEOUT_MSECS = 0
+
+#========================== Syslog wrappers ==========================
+
+def log_info(msg):
+    syslog.openlog()
+    syslog.syslog(syslog.LOG_INFO, msg)
+    syslog.closelog()
+
+def log_warning(msg):
+    syslog.openlog()
+    syslog.syslog(syslog.LOG_WARNING, msg)
+    syslog.closelog()
+
+def log_error(msg):
+    syslog.openlog()
+    syslog.syslog(syslog.LOG_ERR, msg)
+    syslog.closelog()
+
+#========================== Signal Handling ==========================
+
+def signal_handler(sig, frame):
+    if sig == signal.SIGHUP:
+        log_info("Caught SIGHUP - ignoring...")
+        return
+    elif sig == signal.SIGINT:
+        log_info("Caught SIGINT - exiting...")
+        sys.exit(128 + sig)
```

Comment:
not the best solution for common handling of signals

Reply:
The purpose of this package is to provide the common function for the daemons. If any daemon requires another signal handler, it can implement it inside of daemon rather than using this common handler.

Reply:
There is a new PR for xcvrd (https://github.com/Azure/sonic-platform-daemons/pull/23) which changes the behavior of this signal handler. Maybe we should rethink the best way to incorporate a signal handling framework into this base class to allow for more customization.

Reply:
For this PR for xcvrd, it needs to inherit the DaemonBase class and override the signle_handler() function in my opinion.

Reply:
@kevinwangsk I agree we need to find a better way to handle signals. can you please recheck? in any case need to retest the PR as many tests fails.

Reply:
@jleveque So do you think moving back the signal_handler() to each daemon is a better way?

Reply:
I think it's probably best to include a default signal handler in the base class and allow daemons to override the function if the author would like custom signal handling behavior.

Reply:
Yes, this is what the current code did.

Reply:
Understood. I think that's acceptable. If a better solution is proposed in the future we can consider it.